rtgui/labgrid.cc

```diff
@@ -57,67 +59,94 @@ bool LabGridArea::notifyListener()
+LabGridArea::LabGridArea(rtengine::ProcEvent evt, const Glib::ustring &msg, bool enable_low, bool ciexy):
     Gtk::DrawingArea(),
     evt(evt), evtMsg(msg),
     litPoint(NONE),
-    low_a(0.f), high_a(0.f), low_b(0.f), high_b(0.f),
-    defaultLow_a(0.f), defaultHigh_a(0.f), defaultLow_b(0.f), defaultHigh_b(0.f),
+    low_a(0.f), high_a(0.f), low_b(0.f), high_b(0.f), gre_x(0.f), gre_y(0.f), whi_x(0.f), whi_y(0.f),//these variables are used as xy in Ciexy - no change labels
+    defaultLow_a(0.f), defaultHigh_a(0.f), defaultLow_b(0.f), defaultHigh_b(0.f), defaultgre_x(0.f), defaultgre_y(0.f), defaultwhi_x(0.f), defaultwhi_y(0.f),
     listener(nullptr),
     edited(false),
     isDragged(false),
-    low_enabled(enable_low)
+    low_enabled(enable_low),
+    ciexy_enabled(ciexy)
+    
 {
     set_can_focus(false); // prevent moving the grid while you're moving a point
     add_events(Gdk::EXPOSURE_MASK | Gdk::BUTTON_PRESS_MASK | Gdk::BUTTON_RELEASE_MASK | Gdk::POINTER_MOTION_MASK);
     set_name("LabGrid");
     get_style_context()->add_class("drawingarea");
 }
 
-void LabGridArea::getParams(double &la, double &lb, double &ha, double &hb) const
+void LabGridArea::getParams(double &la, double &lb, double &ha, double &hb, double &gx, double &gy, double &wx, double &wy) const
 {
     la = low_a;
     ha = high_a;
     lb = low_b;
     hb = high_b;
+    gx = gre_x;
+    gy = gre_y;
+    wx = whi_x;
+    wy = whi_y;
+ //  printf("la=%f ha=%f lb=%f hb=%f gx=%f gy=%f\n", la, ha, lb, hb, gx, gy);
 }
 
 
-void LabGridArea::setParams(double la, double lb, double ha, double hb, bool notify)
+void LabGridArea::setParams(double la, double lb, double ha, double hb, double gx, double gy, double wx, double wy, bool notify)
 {
     const double lo = -1.0;
     const double hi = 1.0;
     low_a = rtengine::LIM(la, lo, hi);
     low_b = rtengine::LIM(lb, lo, hi);
     high_a = rtengine::LIM(ha, lo, hi);
     high_b = rtengine::LIM(hb, lo, hi);
+    gre_x = rtengine::LIM(gx, lo, hi);
+    gre_y = rtengine::LIM(gy, lo, hi);
+    whi_x = rtengine::LIM(wx, lo, hi);
+    whi_y = rtengine::LIM(wy, lo, hi);
+    
     queue_draw();
     if (notify) {
         notifyListener();
     }
 }
 
-void LabGridArea::setDefault (double la, double lb, double ha, double hb)
+void LabGridArea::setDefault (double la, double lb, double ha, double hb, double gx, double gy, double wx, double wy)
 {
     defaultLow_a = la;
     defaultLow_b = lb;
     defaultHigh_a = ha;
     defaultHigh_b = hb;
+    defaultgre_x = gx;
+    defaultgre_y = gy;
+    defaultwhi_x = wx;
+    defaultwhi_y = wy;
 }
 
 
 void LabGridArea::reset(bool toInitial)
 {
     if (toInitial) {
-        setParams(defaultLow_a, defaultLow_b, defaultHigh_a, defaultHigh_b, true);
+        setParams(defaultLow_a, defaultLow_b, defaultHigh_a, defaultHigh_b, defaultgre_x, defaultgre_y, defaultwhi_x, defaultwhi_y, true);
     } else {
-        setParams(0., 0., 0., 0., true);
+        printf("RESET \n");
```

Comment:
This should be guarded in verbose, or removed

Reply:
I removed